Automatically fix malformed ticket URLs, and improve error message

mnelson4 opened the pull request:
See https://github.com/eventespresso/eea-ticketing/issues/3



<!-- Thanks for your pull request.  Comments within these type of tags will be hidden and not show up when you submit your pull request -->
<!-- Please answer the following questions in order to expediate acceptance -->

## Problem this Pull Request solves
<!-- Please describe your changes in the context of the problem they solve -->
Sometimes folks wound up with malfomed ticket URLs (see https://github.com/eventespresso/eea-ticketing/issues/3#issuecomment-404956901 for the theory on how those happened, but the main takeaway: they had bad URLs, but the problem hopefully won't come up more). When this happened, they'd visit the page and see a message that's pretty hard for the average joe to understand. 
This attempts to fix the URL automatically by changing all `&amp;`s in the querystring into `&`s. 
If that doesn't work, we give a (hopefully) more user-friendly message.

## How has this been tested
<!-- Please describe in detail how you tested your changes and how testing can be reproduced -->
<!-- Include details of your testing environment, and tests ran to see how your changes affect other areas of code -->
<!-- Include any notes about automated tests you've written for this pull request.  Pull requests with automated tests are preferred. -->
* [  ] Register and pay for an event. In the ticket message you receive, click on the link to print tickets. Replace the "&"s with `&amp;`s and try again. You should get a helpful error message. 

## Checklist

* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [x] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [x] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.

nerrad commented on EED_Ticketing.module.php:
```diff
@@ -564,9 +564,15 @@ protected function _generate_tickets($approved_only = false)
             throw new EE_Error(
                 esc_html__(
-                    'The request for the "ee-txn-tickets-url" route has a malformed url.',
+                    'Whoa! Dude! Something went wrong and we\'re unable to display your ticket. Please contact the website administrator and inform them that the ticket URL is invalid.',
```
While I understand the informality, I don't think the language sentiment is appropriate for an error message (will it be always be a  "dude" seeing this message?  Is the tone too informal for some of the sites using EE for their business?).  You might want to consult with @lorenzocaum or others on what to write here instead.  As this is a change in the translated string, we'll want to make sure we carefully consider what's put here so it doesn't have to be changed again.

mnelson4 replied:
@joshfeck thoughts on this? I'm up for whatever

nerrad commented on EED_Ticketing.module.php:
```diff
@@ -564,9 +564,15 @@ protected function _generate_tickets($approved_only = false)
+                . sprintf(
+                    esc_html__('If you\'re comfortable with a little do-it-yourself repairing, you could try changing any "%1$s"s in the address bar to simply "%2$s".', 'event_espresso'),
+                    '&ampamp;',
+                    '&'
```
Instead of suggesting people try repairing themselves (which is bad ux).  What about instead simply providing another link to try instead?  Or better yet, have EE automatically try converting all `&amp;` to `&` and reparsing the url?

mnelson4 replied:
Josh was suggesting that too on the ticket. I was thinking it would be too much work to try to automatically solve it (especially when we're no longer able to reproduce the original issue with SendGrid), but I'll have another look in the code...

mnelson4 replied:
ya I looked in there and it was an easy addition, actually. I incorrectly thought we'd need to refactor some EE core messages code, or the request object. But I think the change was straightforward enough

mnelson4 commented:
@joshfeck @lorenzocaum do you have suggestions on the translated string change here? https://github.com/eventespresso/eea-ticketing/pull/5/files
(I removed the original "Whoa! Dude" as Darren suggested, but we could put it back in. WordPress core has some less formal language too, like the old "Cheatin' huh?". I also removed the extra lines about trying to manually fix the URL, as, after some more investigation, I realized it was possible to fix it for this case really easily)

lorenzocaum commented:
I asked Mike in Slack about the conditions where this would appear and I suggested this message:

```
Something went wrong and we\'re unable to display your ticket. Please contact the event organizer to get your ticket.
```

mnelson4 commented:
In slack @joshfeck said this was fine.
Back for review @nerrad (note we're now lazily automatically trying to fix the URLs)

joshfeck commented:
I tested this branch and it works well.

- replaced the & with `&amp;` and still got a ticket
- replaced & with `&bork;` and got the helpful error message
